src/status_im/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -264,7 +264,32 @@
         limit-insufficient?                         (> (controlled-input/numeric-value input-state)
                                                        current-limit)
         should-try-again?                           (and (not limit-insufficient?) no-routes-found?)
-        current-address                             (rf/sub [:wallet/current-viewing-account-address])]
+        current-address                             (rf/sub [:wallet/current-viewing-account-address])
+        request-fetch-routes                        (fn [bounce-duration-ms]
+                                                      (fetch-routes
+                                                       {:amount             amount-in-crypto
+                                                        :valid-input?       valid-input?
+                                                        :bounce-duration-ms bounce-duration-ms
+                                                        :token              token}))
+        swap-between-fiat-and-crypto                (fn [swap-to-crypto-currency?]
+                                                      (set-just-toggled-mode? true)
+                                                      (set-crypto-currency swap-to-crypto-currency?)
+                                                      (set-input-state
+                                                       (fn [input-state]
+                                                         (controlled-input/set-input-value
+                                                          input-state
+                                                          (let [value     (controlled-input/input-value
+                                                                           input-state)
+                                                                new-value (if swap-to-crypto-currency?
+                                                                            (.toFixed (/ value
+                                                                                         conversion-rate)
+                                                                                      crypto-decimals)
+                                                                            (.toFixed (* value
+                                                                                         conversion-rate)
+                                                                                      12))]
+                                                            (number/remove-trailing-zeroes
+                                                             new-value))))))]
```

Comment:
Maybe we can move this to a separate function so that it has a name and it'll be easier to understand what this is doing, also the code here will look better, wdyt?

Reply:
Well, I totally agree with you but this is not a new function, I just moved handlers to `let` section, so the code down below is easier to comprehend: `:on-swap swap-between-fiat-and-crypto`.
But this let block will benefit from reduction, for sure. I will try to move some functions outside

Reply:
haha, yeah let's start to split this file up somehow 😅